lms/djangoapps/instructor/views/api.py

```diff
@@ -1388,6 +1398,43 @@ def validator(file_storage, file_to_validate):
     return JsonResponse()
 
 
+@method_decorator(transaction.non_atomic_requests, name='dispatch')
+class CohortCSV(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Submit a CSV file to assign users to cohorts
+
+    **Example Requests**:
+
+        POST /api/cohorts/v1/courses/{course_id}/users/
+
+    **Response Values**
+        * Empty as this is executed asynchronously.
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
```

Comment:
Should use `OAuth2AuthenticationAllowInactiveUser` here, also probably `SessionAuthenticationAllowInactiveUser`

Reply:
Right, forgot it didn't reuse the mixing.

Reply:
Not at all, let me fix it.